scripts/utils.ts

```diff
@@ -18,3 +20,67 @@ export const updateContractConfig = (network, newConfig) => {
   fs.writeFileSync(configPath, JSON.stringify(config, null, 2));
   return true;
 };
+
+const getDerivationPathForIndex = (i: number) => `44'/60'/0'/0/${i}`;
+
+export const getLedgerSigner = (index: number, provider: any): LedgerSigner => {
+  const signer = new LedgerSigner(
+    provider,
+    null,
+    getDerivationPathForIndex(index)
+  );
+  // Fix signing for EIP-1559 while ethers.js isn't fixed.
+  signer.signTransaction = ledgerSignTransaction;
+  return signer;
+};
+
+/*
+ * Fixes LedgerSigner for EIP1559 while ethers.js isn't fixed.
+ * The package.json also uses "resolutions" to upgrade the ledger
+ * dependencies to the correct version.
+ */
+export async function ledgerSignTransaction(
+  transaction: ethers.providers.TransactionRequest
+): Promise<string> {
+  const tx = await ethers.utils.resolveProperties(transaction);
+  const baseTx: ethers.utils.UnsignedTransaction = {
+    chainId: tx.chainId || undefined,
```

Comment:
Do we need all these " || undefined"? Won't it just be undefined if tx.chainId (for example) itself is undefined?

Reply:
Shouldn't we check that in the contract then?

Reply:
This is not a contract call, it's submitting a transaction. It overrides the internal implementation in ethers because they're lacking support for the LedgerSigner at the moment.

Reply:
https://github.com/ethers-io/ethers.js/pull/2056/files